examples/kubernetes/README.md

```diff
@@ -0,0 +1,154 @@
+
+## Installation
+Docker (version <= 18.06) and Nvidia plugin installation (version > 2.0)
+
+```
+sudo apt-get install docker-ce=18.03.1~ce-0~ubuntu
+sudo apt-get install nvidia-docker2=2.0.3+docker18.03.1-1 nvidia-container-runtime=2.0.0+docker18.03.1-1
+sudo systemctl daemon-reload
+sudo systemctl restart docker
+```
+
+Test nvidia-docker
+
+`sudo docker run --runtime=nvidia --rm nvidia/cuda:9.0-base nvidia-smi`
+
+Minikube installation. Normally minikube uses a virtual machine with docker inside. We chose to use docker from host not to add additional layer of abstraction.
+
+```
+sudo apt-get update && sudo apt-get install -y apt-transport-https
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+echo "deb https://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee -a /etc/apt/sources.list.d/kubernetes.list
+sudo apt-get update
+sudo apt-get install -y kubectl
+
+curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 \
+  && chmod +x minikube
+
+sudo cp minikube /usr/local/bin && rm minikube
+```
+
+Allow kubernetes to use GPU (Nvidia driver must be ~> 361.93)
+
+Set nvidia-docker as default runtime
+
+In daemon.json which should be located in /etc/docker/ add:
+
+"default-runtime": "nvidia",
+
+Afterwards it should look something like this:
+
+{
+    "default-runtime": "nvidia",
+    "runtimes": {
+        "nvidia": {
+            "path": "/usr/bin/nvidia-container-runtime",
+            "runtimeArgs": []
+        }
+    }
+}
+
+Starting kubernetes through minikube
+
+`sudo minikube start --vm-driver=none`
+
+When minikube tells us everything look all right we need to deploy nvidia plugin allowing to use GPUs on our node(s)
+
+```
+sudo kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/v1.11/nvidia-device-plugin.yml
+```
+
+Restart docker and kubernetes:
+```
+sudo minikube stop
+sudo service docker stop
+sudo service docker start
+sudo minikube start --vm-driver=none
+```
+
+Microk8s installation and configuration.
+
+```
+sudo snap install microk8s --classic
+sudo microk8s.start
+microk8s.kubectl get nodes
+microk8s.enable gpu
+sudo microk8s.stop
+sudo microk8s.start
+```
+
+As shown microk8s is easier and require less configuration. There is also quite a lot of built in addons that can be turned on the same way as gpu one. Due to the usage of snap, unistallation is easier and there is no need to worry about leftovers. Micro8s also comes with its own kubectl so it won't after existing configuration if there is any but can be also aliased to be used with kubectl command. 
+
+`sudo kubectl get nodes`
```

Comment:
To see available nodes:
`sudo kubectl get nodes`

To see more info about a node, pod or job:
`sudo kubectl describe node/pod/job <name>`
Example:
`sudo kubectl describe node minikube`
Somewhere in `capactity` it should output `nvidia.com/gpu: X` with X being number of available GPUs on this machine.

If everything worked correctly, now we should be able to create a pod using yaml template file from one of our examples:
1. Download `Dockerfile` and build image from it
2. Download template `sh` script, optionally adjust image name (`image:`) and run it. 

### Running training examples
To see more detailed instructions on how to run specific training example check out [DeepSpeech README](https://github.com/roscisz/TensorHive/tree/kubernetes/examples/kubernetes/deepspeech) and [T2T README](https://github.com/roscisz/TensorHive/tree/kubernetes/examples/kubernetes/t2t_transformer)

<This should probably go directly into that single, specific README>
Template will create a training job for Mozilla's Deepspeech using a single GPU.

Commands to try out:
```
# Replace <name> with: 
# - ds (DeepSpeech)
# - t2t (T2T)
sudo kubectl get pod<name>
sudo kubectl logs pod <name>
sudo kubectl describe pod <name>
sudo kubectl delete pod <name>
```